File: .cpp/Program/Laptop.cpp
#pragma once
#include <bits/stdc++.h>
#include "Computer.cpp"
using namespace std;

class Laptop : public Computer {
    private:
        int batteryCapacity;
        double screenSize;
        double weight;

    public:
        Laptop () { }

        Laptop (string brand, string model, string OS, CPU cpu, vector<RAM> ram, vector<Storage*> storage, int batteryCapacity, double screenSize, double weight) : Computer (brand, model, OS, cpu, ram, storage) {
            this->batteryCapacity = batteryCapacity;
            this->screenSize = screenSize;
            this->weight = weight;
        }

        void setBatteryCapacity (int batteryCapacity) { this->batteryCapacity = batteryCapacity; }
        void setScreenSize (int screenSize) { this->screenSize = screenSize; }
        void setWeight (int weight) { this->weight = weight; }

        int getBatteryCapacity () { return batteryCapacity; }
        double getScreenSize () { return screenSize; }
        double getWeight () { return weight; }

        void info () override {
            cout << "Laptop" << endl;
            Computer::info();
            cout << "| Battery Capacity  : " << batteryCapacity << "mAh" << endl;
            cout << "| Screen Size       : " << screenSize << " inch" << endl;
            cout << "| Weight            : " << weight << "g" << endl;
        }

        ~Laptop () { }
};

File: .py/Program/Laptop.py
from Computer import Computer

class Laptop (Computer):
    __batteryCapacity = 0
    __screenSize = 0.0
    __weight = 0.0

    def __init__ (self, brand = "", model = "", OS = "", CPU = None, RAM = None, Storage = None, batteryCapacity = 0, screenSize = 0.0, weight = 0.0):
        super ().__init__ (brand, model, OS, CPU, RAM, Storage)
        self.__batteryCapacity = batteryCapacity
        self.__screenSize = screenSize
        self.__weight = weight

    def setBatteryCapacity (self, batteryCapacity):
        self.__batteryCapacity = batteryCapacity

    def setScreenSize (self, screenSize):
        self.__screenSize = screenSize

    def setWeight (self, weight):
        self.__weight = weight

    def getBatteryCapacity (self):
        return self.__batteryCapacity
    
    def getScreenSize (self):
        return self.__screenSize
    
    def getWeight (self):
        return self.__weight
    
    def info (self):
        print ("| Laptop")
        super ().info ()
        print (f"| Battery Capacity  : { self.__batteryCapacity }mAh")
        print (f"| Screen Size       : { self.__screenSize } inch")
        print (f"| Weight            : { self.__weight }g")